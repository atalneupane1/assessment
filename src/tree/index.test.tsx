import React from "react";
import Tree from "./index";
import { render } from "@testing-library/react";

// Mocking a simple snapshot test
test("it matches the snapshot", () => {
  const { asFragment } = render(<Tree />);
  expect(asFragment()).toMatchSnapshot();
});