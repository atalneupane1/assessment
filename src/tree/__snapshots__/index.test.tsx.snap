// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="tree"
  >
    <button>
      Alphabetize
    </button>
    <div
      class="node"
    >
      root
      <button>
        ❌
      </button>
      <div
        class="children"
      >
        <div
          class="node"
        >
          a.nt
          <button>
            ❌
          </button>
        </div>
        <div
          class="node"
        >
          b.ear
          <button>
            ❌
          </button>
          <div
            class="children"
          >
            <div
              class="node"
            >
              c..at
              <button>
                ❌
              </button>
            </div>
            <div
              class="node"
            >
              d..og
              <button>
                ❌
              </button>
              <div
                class="children"
              >
                <div
                  class="node"
                >
                  e...lephant
                  <button>
                    ❌
                  </button>
                </div>
                <input
                  class="children"
                  placeholder="Add child"
                  type="text"
                />
              </div>
            </div>
            <input
              class="children"
              placeholder="Add child"
              type="text"
            />
          </div>
        </div>
        <div
          class="node"
        >
          f.rog
          <button>
            ❌
          </button>
          <div
            class="children"
          >
            <div
              class="node"
            >
              d..olphin
              <button>
                ❌
              </button>
            </div>
            <input
              class="children"
              placeholder="Add child"
              type="text"
            />
          </div>
        </div>
        <input
          class="children"
          placeholder="Add child"
          type="text"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;